test/persist/test_failure_indices.sh

```diff
@@ -0,0 +1,124 @@
+#! /bin/bash
+
+export PATH="$GOBIN:$PATH"
+export TMHOME=$HOME/.tendermint_persist
+
+rm -rf "$TMHOME"
+tendermint init
+
+# use a unix socket so we can remove it
+RPC_ADDR="$(pwd)/rpc.sock"
+
+TM_CMD="tendermint node --log_level=debug --rpc.laddr=unix://$RPC_ADDR" # &> tendermint_${name}.log"
+DUMMY_CMD="abci-cli kvstore --persist $TMHOME/kvstore" # &> kvstore_${name}.log"
+
+
+function start_procs(){
+    name=$1
+    indexToFail=$2
+    echo "Starting persistent kvstore and tendermint"
+    if [[ "$CIRCLECI" == true ]]; then
+        $DUMMY_CMD &
+    else
+        $DUMMY_CMD &> "kvstore_${name}.log" &
+    fi
+    PID_DUMMY=$!
+
+    # before starting tendermint, remove the rpc socket
+    rm -f $RPC_ADDR
+    if [[ "$indexToFail" == "" ]]; then
+        # run in background, dont fail
+		if [[ "$CIRCLECI" == true ]]; then
+			$TM_CMD &
+		else
+            $TM_CMD &> "tendermint_${name}.log" &
+		fi
+        PID_TENDERMINT=$!
+    else
+        # run in foreground, fail
+		if [[ "$CIRCLECI" == true ]]; then
+			FAIL_TEST_INDEX=$indexToFail $TM_CMD
+		else
+            FAIL_TEST_INDEX=$indexToFail $TM_CMD &> "tendermint_${name}.log"
+		fi
+        PID_TENDERMINT=$!
+    fi
+}
+
+function kill_procs(){
+    kill -9 "$PID_DUMMY" "$PID_TENDERMINT"
```

Comment:
I think SIGTERM should be sufficient.

```suggestion
    kill -TERM "$PID_DUMMY" "$PID_TENDERMINT"
```